ZoomZoom/Public/Get-OutlookCalendarAppointment.ps1

```diff
@@ -0,0 +1,100 @@
+.EXAMPLE
+    Get-OutlookCalendarAppointments
+    Returns Outlook calendar appointments for the next 48 hours
+
+    Get-OutlookCalendarAppointments -Start 12/01/2020 -End 21/02/2020
+    Returns all non-recurring meetings between 12th January and 21st February 2020
+
+.OUTPUTS
+    PSCustomObject
+.NOTES
+
+#>
+function Get-OutlookCalendarAppointment {
+
+    [CmdletBinding(DefaultParameterSetName = 'Default')]
+
+    param (
+
+        # The start date for the search
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ParameterSetName = 'SearchTime')]
+        $Start,
+
+        # The end date for the search
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ParameterSetName = 'SearchTime')]
+        $End
+    )
+
+    begin {
+        #Add the required Assembly to PowerShell before processing input
+        Add-Type -assembly "Microsoft.Office.Interop.Outlook"
+
+        #Defines the DefaultFolderID for the Calendar, and creates our new COM Object
+        $OutlookFolders = 'Microsoft.Office.Interop.Outlook.OlDefaultFolders' -as [type]
+        $Outlook = New-Object -ComObject Outlook.Application
+        $NameSpace = $Outlook.GetNamespace('MAPI')
+
+        #Empty array to store our results
+        $Meetings = @()
+        $Results = @()
+        if ($PSCmdlet.ParameterSetName -eq 'Default') {
+            $Start = Get-Date
+            $End = $Start.AddDays(2)
+        }
+    }
+
+    process {
+        $Folder = $NameSpace.getDefaultFolder($OutlookFolders::olFolderCalendar)
+        $RecurringAppointments = $Folder.Items | Where-Object -Property IsRecurring -eq $True
+        if ($null -eq $start) {
+            $Meetings += $Folder.Items | Where-Object -Property Start -gt ((Get-Date -Hour 00 -Minute 00 -Second 00).AddDays(-1))
+        }
+        else {
+            $Meetings += $Folder.Items | Where-Object -Property Start -gt (Get-Date $Start -Hour 00 -Minute 00 -Second 00) | Where-Object -Property End -lt (Get-Date $End -Hour 00 -Minute 00 -Second 00)
+        }
+    }
+
+    end {
+        foreach ($Meeting in $Meetings) {
+            $Results += [PSCustomObject]@{
+                MeetingTime    = $Meeting.Start
+                MeetingSubject = $Meeting.Subject
+                MeetingBody    = $Meeting.Body
+                MeetingURL     = Get-ZoomStringFromBody $Meeting.Body
+                Recurring      = $false
+            }
+        }
+        if ($Null -eq $RecurringAppointments) {
+            return $Results
+        }
+        else {
+            Foreach ($Appointment in $RecurringAppointments) {
+                $AppointmentDates = @()
+                $AppointmentDates += (Get-RecurringMeetingDates -Appointment $Appointment -EndDate (Get-Date $End).ToString()).AppointmentDates
+                foreach ($Date in $AppointmentDates) {
+                    if ($Date -gt (Get-Date $Start)) {
+                        $Results += [pscustomobject]@{
```

Comment:
Can we specify the default display set for the properties of the custom object doing it this way?
The fact that the meeting is recurring is not directly important, so hidden using the property set created before results are returned. It is a potentially helpful property to have however

Lines 95-98 for this block in question.

Reply:
What if you take 95-98 and move it to this `foreach` (inside the `if` on 83), and then pass to the pipeline after you've set the properties?

Will PowerShell care if you pass an object with default display set, rather than an array of objects with default display set? I have a feeling it'll just merge the objects, as long as the display set is the same.